discojs/src/aggregator/get.ts

```diff
@@ -1,33 +1,60 @@
 import type { Task } from '../index.js'
 import { aggregator } from '../index.js'
+import type { Model } from "../index.js";
 
-/**
- * Enumeration of the available types of aggregator.
- */
-export enum AggregatorChoice {
-  MEAN,
-  SECURE,
-  BANDIT
+type AggregatorOptions = {
+  model?: Model,
+  roundCutOff?: number, // MeanAggregator
+  threshold?: number, // MeanAggregator
+  thresholdType?: 'relative' | 'absolute', // MeanAggregator
 }
 
 /**
- * Provides the aggregator object adequate to the given task.
- * @param task The task
+ * Initializes an aggregator according to the task definition, the training scheme and the specified options.
+ * Here is the ordered list of parameters used to define the aggregator's default behavior:
+ * task.trainingInformation.aggregator > scheme > task.trainingInformation.scheme
+ * 
+ * If `task.trainingInformation.aggregator` is defined, we initialize the chosen aggregator with `options`.
+ * 
+ * Otherwise, if `scheme` is defined, we initialize a MeanAggregator for both training schemes:
+ * For federated learning or local training the aggregator waits for a single contribution to trigger a model update.
+ * (the server's model update for federated learning or our own contribution if training locally)
+ * For decentralized learning the aggregator waits for every nodes' contribution to trigger a model update.
+ * 
+ * If `scheme` is undefined, we rely on task.trainingInformation.scheme to infer `scheme`.
+ * 
+ * @param task The task object associated with the current training session
+ * @param scheme The training scheme. If not specified, the task training information scheme is used
+ * @param options Options passed down to the aggregator's constructor
  * @returns The aggregator
  */
-export function getAggregator (task: Task): aggregator.Aggregator {
-  const error = new Error('not implemented')
+export function getAggregator(task: Task, scheme?: Required<Task['trainingInformation']['scheme']>,
+  options?: AggregatorOptions): aggregator.Aggregator {
+  if (options === undefined) {
+    options = {} // init empty arguments if not specified
+  }
```

Comment:
default arguments behave in the same way

```suggestion
  options: AggregatorOptions = {}): aggregator.Aggregator {
```